saleor/order/utils.py

```diff
@@ -42,16 +42,20 @@ def cancel_order(order):
 def recalculate_order(order):
     """Recalculate and assign total price of order.
 
-    Total price is a sum of items in shipment groups and order shipping price.
+    Total price is a sum of items in shipment groups and order shipping price
+    minus discount amount.
     """
     prices = [
         group.get_total() for group in order
         if group.status != GroupStatus.CANCELLED]
-    total_net = sum(p.net for p in prices)
-    total_gross = sum(p.gross for p in prices)
-    total = Price(
-        net=total_net, gross=total_gross, currency=settings.DEFAULT_CURRENCY)
-    total += order.shipping_price
+    total_net = sum(p.net.amount for p in prices)
+    total_gross = sum(p.gross.amount for p in prices)
+    total = TaxedMoney(
+        net=Money(total_net, currency=settings.DEFAULT_CURRENCY),
+        gross=Money(total_gross, currency=settings.DEFAULT_CURRENCY))
+    total += TaxedMoney(net=order.shipping_price, gross=order.shipping_price)
+    if order.discount_amount:
+        total -= order.discount_amount
```

Comment:
We update here total decreasing total by discount_amount. But what if order has applied voucher "5 USD discount for each book" and staff user has just increased quantity of ordered books?

Reply:
I think it's safer if we allowed staff to manually adjust prices after modifying an order. The original discount could no longer exist, could no longer apply (because of date ranges, because of usage limits, because a qualifying row was removed from the order etc.) and so on.

Reply:
Recounting the orders based on current or past state of the store configuration is something that I've been discussing with @elwoodxblues for some time now. Last time we've talked about this it felt to us that no mechanic may be smarter here than human modifying the order and thus it should be up to human handling the editing to be able to change discount's amount based on circumstances.